core/sdk/module_client_generator.go

```diff
@@ -41,6 +51,16 @@ func (sdk *clientGeneratorModule) GenerateClient(
 	deps *core.ModDeps,
 	outputDir string,
 ) (inst dagql.ObjectResult[*core.Directory], err error) {
+	query, err := core.CurrentQuery(ctx)
+	if err != nil {
+		return inst, err
+	}
+	dagqlCache, err := query.Cache(ctx)
+	if err != nil {
+		return inst, err
+	}
+	dag := sdk.mod.dag.WithCache(dagqlCache)
```

Comment:
Maybe struggling to understand what this is doing - it's putting the right session cache in, but how can it end up being the wrong one?

Might be worth splitting it out to a `sdk.mod.Dag(ctx)` or something?